app/vmagent/remotewrite/client.go

```diff
@@ -459,6 +458,35 @@ again:
 		_ = resp.Body.Close()
 		c.packetsDropped.Inc()
 		return true
+	} else if statusCode == http.StatusUnsupportedMediaType {
+		if c.useVMProto.Swap(false) {
+			logger.Infof("received unsupported media type from remote storage at %q. Downgrading protocol from VictoriaMetrics to Prometheus remote write for all future requests. "+
+				"See https://docs.victoriametrics.com/vmagent/#victoriametrics-remote-write-protocol", c.sanitizedURL)
+		}
+
+		if encoding.IsZstd(block) {
+			logger.Infof("received unsupported media type from remote storage at %q. Re-packing the block to Prometheus remote write and retrying."+
+				"See https://docs.victoriametrics.com/vmagent/#victoriametrics-remote-write-protocol", c.sanitizedURL)
+
+			block, err = repackBlockFromZstdToSnappy(block)
+			if err != nil {
+				logger.Errorf("cannot re-pack block with size %d bytes from Zstd to Snappy for %q: %s; the block will be dropped",
```

Comment:
I'd make this `logger.Panicf("BUG: ...")`, since this error may occur only in two cases:

- Programming error (aka BUG) when a block with improper encoding has been put in the persistent queue
- Data corruption when the block has been corrupted in the persistent queue.

Both cases need immediate attention from operations in order to fix them ASAP, so the best way to do this is to panic with the information, which help fixing the issue quickly.